### PR TITLE
RDKVREFPLT-3158: Split iarmmgrs hal into individual components

### PR DESCRIPTION
**Pull request by Mallikarjunm6316**

### CODE REVIEW

**Review by joseinweb — COMMENTED** [review]
This should be 93bd52394d371a50b7265cc24b3fa43e2fed346

File: recipes-halif/deepsleepmgr/deepsleepmgr-hal-raspberrypi4.bb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+DESCRIPTION = "deepsleep HAL Implementation."
+LICENSE = "Apache-2.0"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=b1e01b26bacfc2232046c90a330332b3"
+
+PROVIDES = "virtual/deepsleep-hal virtual/vendor-deepsleep-hal"
+RPROVIDES_${PN} += " virtual/vendor-deepsleepmgr-hal "
+
+SRC_URI = "git://github.com/Mallikarjunm6316/rdkvhal-deepsleep-manager-raspberrypi4.git;branch=develop;protocol=https"
+
+SRCREV = "f08a3a71e76e2c3f50567095ac470b5bcb18f6ad"
```
Comment (joseinweb) [inline recipes-halif/deepsleepmgr/deepsleepmgr-hal-raspberrypi4.bb:10]: This should be 93bd52394d371a50b7265cc24b3fa43e2fed346

File: recipes-halif/pwrmgr/pwrmgr-hal-raspberrypi4.bb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+DESCRIPTION = "PWRMGR HAL Implementation."
+LICENSE = "Apache-2.0"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=b1e01b26bacfc2232046c90a330332b3"
+
+PROVIDES += " virtual/vendor-pwrmgr-hal "
+RPROVIDES_${PN} += " virtual/vendor-pwrmgr-hal"
+
+SRC_URI = "git://github.com/Mallikarjunm6316/rdkvhal-power-manager-raspberrypi4.git;branch=develop;protocol=https"
+
+SRCREV = "d164a5fe0d67f1ce8c8fddaf631e561c1e744466"
```
Comment (joseinweb) [inline recipes-halif/pwrmgr/pwrmgr-hal-raspberrypi4.bb:10]: This should be f174dc2bec24db8d19a2b384c6dccfaaa530c775

**Review by joseinweb — CHANGES_REQUESTED** [review]

**Review by deepthi-ps — CHANGES_REQUESTED** [review]
1- Remove VENDOR_IPK_SERVER_PATH from meta-rdk-raspberrypi. It should be only in vendor release layer.
2 - Change SRC_URI from your local repo to rdkcentral repo.
3 - If you remove virtual/vendor-iarmmgrs-hal, how will middleware build pass ?

File: recipes-halif/deepsleepmgr/deepsleepmgr-hal-raspberrypi4.bb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+#SRCREV = "b079260ee057fb02f0bef2c8201cb2c2141c8663"
+
+S = "${WORKDIR}/git"
+
+DEPENDS = "iarmbus-headers deepsleep-manager-headers"
+
+inherit autotools coverity
+
+IMAGE_INSTALL += "lib32-deepsleepmgr-hal-raspberrypi4"
```
Comment (arun-madhavan-013) [inline recipes-halif/deepsleepmgr/deepsleepmgr-hal-raspberrypi4.bb:20]: Please move this to the right recipe.

File: recipes-halif/deepsleepmgr/deepsleepmgr-hal-raspberrypi4.bb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+DESCRIPTION = "deepsleep HAL Implementation."
+LICENSE = "Apache-2.0"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=b1e01b26bacfc2232046c90a330332b3"
+
+PROVIDES = "virtual/deepsleep-hal virtual/vendor-deepsleep-hal"
+RPROVIDES_${PN} += " virtual/vendor-deepsleepmgr-hal "
+
+SRC_URI = "${CMF_GITHUB_ROOT}/rdkvhal-deepsleep-manager-raspberrypi4;${CMF_GIT_SRC_URI_SUFFIX}"
+
+#SRC_URI = "https://github.com/Mallikarjunm6316/rdkvhal-deepsleep-manager-raspberrypi4;sha256sum=cf7c8fd1c7a0a56f662ea8ed6d6ab6aab7b4483e0a28a23bb463d1acf2c1bbbc;branch=develop;protocol=https"
```
Comment (arun-madhavan-013) [inline recipes-halif/deepsleepmgr/deepsleepmgr-hal-raspberrypi4.bb:10]: Clean up the unnecessary entries.

File: recipes-halif/pwrmgr/pwrmgr-hal-raspberrypi4.bb [hunk 5]
```diff
@@ -0,0 +1,22 @@
+DESCRIPTION = "PWRMGR HAL Implementation."
+LICENSE = "Apache-2.0"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=b1e01b26bacfc2232046c90a330332b3"
+
+PROVIDES += " virtual/vendor-pwrmgr-hal "
+RPROVIDES_${PN} += " virtual/vendor-pwrmgr-hal"
+
+SRC_URI = "${CMF_GITHUB_ROOT}/rdkvhal-power-manager-raspberrypi4;${CMF_GIT_SRC_URI_SUFFIX}"
+
+#SRC_URI = "https://github.com/Mallikarjunm6316/rdkvhal-power-manager-raspberrypi4;sha256sum=e214e6664a4a19f58b21b6074c0e272f92ba7ef5c05c2930eb161ccc75c14738;ranch=develop;protocol=https"
```
Comment (arun-madhavan-013) [inline recipes-halif/pwrmgr/pwrmgr-hal-raspberrypi4.bb:10]: Clean-up if not required.

File: conf/include/vendor_pkg_versions.inc [hunk 6]
```diff
@@ -192,10 +192,20 @@ PV_pn-hdmicec-hal-raspberrypi4 = "1.0.0"
 PR_pn-hdmicec-hal-raspberrypi4 = "r0"
 PACKAGE_ARCH_pn-hdmicec-hal-raspberrypi4 = "${VENDOR_LAYER_EXTENSION}"
 
-SRCREV_pn-iarmmgrs-hal-raspberrypi4 = "1.0.1"
-PV_pn-iarmmgrs-hal-raspberrypi4 = "2.0.1"
-PR_pn-iarmmgrs-hal-raspberrypi4 = "r0"
-PACKAGE_ARCH_pn-iarmmgrs-hal-raspberrypi4 = "${VENDOR_LAYER_EXTENSION}"
+#SRCREV_pn-iarmmgrs-hal-raspberrypi4 = "1.0.0"
```
Comment (arun-madhavan-013) [inline conf/include/vendor_pkg_versions.inc:195]: Clean up if not required.

**Review by arun-madhavan-013 — CHANGES_REQUESTED** [review]

**Review by deepthi-ps — APPROVED** [review]

File: recipes-halif/deepsleepmgr/deepsleepmgr-hal-raspberrypi4.bb (hunk 3 above)
Comment (arun-madhavan-013) [reply recipes-halif/deepsleepmgr/deepsleepmgr-hal-raspberrypi4.bb:16]: IMAGE_INSTALL += "lib32-deepsleepmgr-hal-raspberrypi4" is not required.
Please remove

**Review by arun-madhavan-013 — CHANGES_REQUESTED** [review]

File: recipes-halif/pwrmgr/pwrmgr-hal-raspberrypi4.bb (hunk 5 above)
Comment (Mallikarjunm6316) [reply recipes-halif/pwrmgr/pwrmgr-hal-raspberrypi4.bb]: Addressed, removed

File: recipes-halif/pwrmgr/pwrmgr-hal-raspberrypi4.bb (hunk 2 above)
Comment (Mallikarjunm6316) [reply recipes-halif/pwrmgr/pwrmgr-hal-raspberrypi4.bb]: Now it's pointing to main repo, so removed the src-rev.

**Comment by Mallikarjunm6316** [comment]
> This should be 93bd52394d371a50b7265cc24b3fa43e2fed346

now pointing to main repo, so src rev is not required.

**Comment by Mallikarjunm6316** [comment]
By mistake closed the PR, so reopening the same

**Comment by Mallikarjunm6316** [comment]
> This should be 93bd52394d371a50b7265cc24b3fa43e2fed346
Now src repo is pointing to main repo, so src rev is not required

**Review by arun-madhavan-013 — APPROVED** [review]

**Review by joseinweb — APPROVED** [review]

**Pull request merged**